pypodcastparser/Item.py

```diff
@@ -207,16 +201,57 @@ def set_guid(self, tag):
             self.guid = tag.string
         except AttributeError:
             self.guid = None
-
+#TODO convert to one timezone
     def set_published_date(self, tag):
         """Parses published date and set value."""
         try:
             self.published_date = tag.string
             self.published_date_string = tag.string
-            pubdate = parse(self.published_date)
-            self.published_date = datetime.datetime.strftime(pubdate, "%Y-%d-%m, %H:%M:%S")
+
+            deconstructed_date = self.published_date_string.split(" ")
+
+
+            if(re.match("^[a-zA-Z]{3}$",deconstructed_date[-1])):
+                deconstructed_date.pop()
+
+            regex_array = ["^[a-zA-Z]{3},$","^\d{1,2}$","^[a-zA-Z]{3}$","^\d{4}$","^\d\d:\d\d"]
+            new_array = []
+
+
+            for i,v in enumerate(deconstructed_date):
+                if(re.match(regex_array[i],v)):
+                    new_array.append(v)
+                else:
+                    for x,z in enumerate(deconstructed_date):
```

Comment:
Not to be too much of a stickler but can we have better variable names other than "x,z"

Reply:
updated in poonams pr